Bookish/Bookish.DataAccess/LibraryService.cs

```diff
@@ -13,5 +21,40 @@ public IEnumerable<Book> GetBooks()
         {
             return databaseConnection.Query<Book>("SELECT * FROM Books");
         }
+
+        public Book GetCopies(string isbn)
+        {
+            var query =
+                "SELECT LibraryBooks.isbn, Books.title, Books.authors, LibraryBooks.bookId, Users.username, Loans.dueDate " +
+                "FROM LibraryBooks " +
+                "JOIN Books ON LibraryBooks.isbn=Books.isbn " +
+                "LEFT JOIN Loans ON LibraryBooks.bookId=Loans.bookId " +
+                "LEFT JOIN Users ON Users.userId=Loans.userId " +
+                $"WHERE LibraryBooks.isbn='{isbn}';";
+
+            var bookDictionary = new Dictionary<string, Book>();
+
+
+            var bookWithCopies = databaseConnection.Query<Book, BookCopy, Book>(
+                    query,
+                    (book, bookCopy) =>
+                    {
+                        Book bookEntry;
+
+                        if (!bookDictionary.TryGetValue(book.isbn, out bookEntry))
```

Comment:
This can be simplified to checking if your book variable is null and setting it to `book` if so 🙂